### PR TITLE
Sync User's account to Troly after Placing Order

### PR DESCRIPTION
**Pull request by rcdtroly**
### Why:
* When sending an order from WC, we need to also send shipping information if it is different to what we have on file.

### What:
* Trigger `user_import` in `woocommerce_checkout_order_processed` hook

### Result:

alt text
### Refs:

https://app.asana.com/0/939825968139797/1124744275741748

### CODE REVIEW

File: wp99234.php [hunk 1]
```diff
@@ -581,10 +581,62 @@ function WP99234() {
...
+        $payload = array(
+            "ID" => $user_id,
+            'billing_address_1' => $posted_data['billing_address_1'],
+            'billing_city' => $posted_data['billing_city'],
+            'billing_postcode' => $posted_data['billing_postcode'],
+            'billing_state' => $posted_data['billing_state'],
+            'billing_country' => $posted_data['billing_country'],
+            'ship_to_different_address' => $posted_data['ship_to_different_address'],
+            'shipping_address_1' => $posted_data['shipping_address_1'],
+            'shippinzg_city' => $posted_data['shipping_city'],
```
Comment (JamesEmpireOne) [inline wp99234.php]: Typo

File: wp99234.php [hunk 2]
```diff
@@ -581,10 +581,63 @@ function WP99234() {
...
+                'shipping_city' => $posted_data['shipping_city'],
+                'shipping_postcode' => $posted_data['shipping_postcode'],
+                'shipping_state' => $posted_data['shipping_state'],
+                'shipping_country' => $posted_data['shipping_country']
+            );
+            $user_data = wp_update_user( $payload );
+
+            if ( is_wp_error( $user_data ) ) {
+                // There was an error; possibly this user doesn't exist.
+                throw new Exception(__( 'Error! Updating user!', 'wp99234') );
```
Comment (ricardo-troly-io) [inline wp99234.php]: Could we manage exceptions with more accurate information such as ids and detailed troubleshooting?

File: wp99234.php [hunk 3]
```diff
@@ -581,10 +581,63 @@ function WP99234() {
...
+            }
+        }
+
+        // Export User account with updated shipping address
+        $results = WP99234()->_users->export_user( $user_id, 'shipping', array(), true );
+        $errors = is_object($results) ? (array)$results->errors : null;
+
+        if ( ! empty( $errors ) ) {
+            throw new Exception( __( 'An error has occurred, saving User upon placing order. ', 'wp99234' ) );
```
Comment (ricardo-troly-io) [inline wp99234.php]: Same as above.

File: wp99234.php [hunk 4]
```diff
@@ -581,10 +581,63 @@ function WP99234() {
...
+
+            ob_start();
+            var_dump( $errors );
```
Comment (ricardo-troly-io) [inline wp99234.php]: Remove var_dump()

File: wp99234.php (hunk 2 above)
Comment (rcdtroly) [reply wp99234.php]: @ricardo-troly-io updated the logging

File: wp99234.php (hunk 3 above)
Comment (rcdtroly) [reply wp99234.php]: @ricardo-troly-io updated the logging

File: wp99234.php (hunk 4 above)
Comment (rcdtroly) [reply wp99234.php]: Fixed

**Pull request merged**